### PR TITLE
Make improvements to report tool 

### PR DESCRIPTION
**Pull request by mebristo**
Make improvements to report tool including:
- allow lists of tables
- add optional base64 encoding of images
- add method to zip report folder

Please follow the guidelines for PRs contained [here](docs/source/contributing.md). Checklist:

- [ ] Ensure that your PR is small, and implements one change.
- [ ] Add unit tests for all functions that you introduced or modified.
- [ ] Run PyCharm's code cleanup tools on your Python files.
- [ ] Ensure that documentation renders correctly in Sphinx (run Sphinx via `make html` in the `docs folder)
- [ ] Link the correct GitHub issue for tracking.
- [ ] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [ ] When merging your PR, **replace the default merge message** with a brief description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

File: hi-ml/src/health_ml/utils/reports.py [hunk 1]
```diff
@@ -149,47 +150,67 @@ def add_text(self, text: str, tag_class: str = '') -> None:
...
-        relative path will be used. This is to ensure that the HTML document is able to locate and embed the image.
-        Otherwise, the image path is not altered.
+        Given a path to an image, add it to the report template and to the report arguments
+        for rendering later. Optionally encode as base64 - this is useful if a standalone
+        report is required but leads to a much larger report file size, so is False by default
 
-        :param image_path: The path to the image to be embedded
-        :param figsize: The size of the figure
+        :param image_path: The paths to the image to be added
+        :param base64_encode: If True, encode the image as base64 in the HTML report. Default is False
```
Comment (vale-salvatelli) [inline hi-ml/src/health_ml/utils/reports.py:208]: this sounds as you have tried to encode the tiles images but this was leading to a too large file - is it true?

File: hi-ml/src/health_ml/utils/reports.py [hunk 2]
```diff
@@ -440,9 +484,26 @@ def render(self, save_html: bool = True) -> None:
...
+
+        :return: The path to the zipped folder
+        """
+        import zipfile
+        report_files = self.report_folder.rglob("*.*")
+        zipped_folder_path = self.report_folder.with_suffix(".zip")
+        with zipfile.ZipFile(zipped_folder_path, "w") as zipped_folder:
+            for report_file in report_files:
+                zipped_folder.write(report_file, arcname=report_file.relative_to(self.report_folder))
+        print(f"Zipped folder path: {str(zipped_folder_path)}")
```
Comment (Shruthi42) [inline hi-ml/src/health_ml/utils/reports.py:508]: Should this be written to a logger instead?

**Review by Shruthi42 — APPROVED** [review]

**Review by vale-salvatelli — DISMISSED** [review]

File: hi-ml/src/health_ml/utils/reports.py (hunk 1 above)
Comment (mebristo) [reply hi-ml/src/health_ml/utils/reports.py:208]: It does lead to pretty large files -the histopathology 1st gen report with encoded images is about 84MB without any compression with encoded images

File: hi-ml/src/health_ml/utils/reports.py (hunk 2 above)
Comment (mebristo) [reply hi-ml/src/health_ml/utils/reports.py:507]: Potentially, but I propose to do this in a future PR!

**Review by vale-salvatelli — APPROVED** [review]

**Pull request merged**